movement/watch_faces/complication/deadline_face.c

```diff
@@ -0,0 +1,593 @@
+
+    /* Correct errors of naive difference */
+    for (i = 0; i < 6; i++) {
+        if (unit[i] < 0) {
+            /* Correct remaining units */
+            if (i == 3)
+                unit[i] += _days_in_month(deadline.unit.month - 1, deadline.unit.year);
+            else
+                unit[i] += range[i];
+
+            /* Carry over change to next unit if non-zero */
+            if (i < 5 && unit[i + 1] != 0)
+                unit[i + 1] -= 1;
+        }
+    }
+
+    /* Set range */
+    i = state->current_index + 1;
+    if (unit[5] > 0) {
+        /* years:months */
+        sprintf(buf, "DL%2d%02d%02dYR", i, unit[5] % 100, unit[4] % 12);
+    } else if (unit[4] > 0) {
+        /* months:days */
+        sprintf(buf, "DL%2d%02d%02dMO", i, (unit[5] * 12 + unit[4]) % 100, unit[3] % 32);
+    } else if (unit[3] > 0) {
+        /* days:hours */
+        sprintf(buf, "DL%2d%02d%02ddY", i, unit[3] % 32, unit[2] % 24);
+    } else {
+        /* hours:minutes:seconds */
+        sprintf(buf, "DL%2d%02d%02d%02d", i, unit[2] % 24, unit[1] % 60, unit[0] % 60);
+    }
+    watch_display_string(buf, 0);
+}
+
+/* Init running mode */
+static void _running_init(movement_settings_t *settings, deadline_state_t *state)
+{
+    (void) settings;
+    (void) state;
+
+    watch_clear_indicator(WATCH_INDICATOR_24H);
+    watch_clear_indicator(WATCH_INDICATOR_PM);
+    watch_set_colon();
+}
+
+/* Loop of running mode */
+static bool _running_loop(movement_event_t event, movement_settings_t *settings, void *context)
+{
+    deadline_state_t *state = (deadline_state_t *) context;
+    _running_display(event, settings, state);
+
+    switch (event.event_type) {
+        case EVENT_ALARM_BUTTON_UP:
+            _beep_button(settings);
+            state->current_index = (state->current_index + 1) % DEADLINE_FACE_DATES;
+            _running_display(event, settings, state);
+            break;
+        case EVENT_ALARM_LONG_PRESS:
+            _beep_enable(settings);
+            _setting_init(settings, state);
+            state->mode = DEADLINE_FACE_SETTING;
+            break;
+        case EVENT_MODE_BUTTON_UP:
+            movement_move_to_next_face();
+            return false;
+        case EVENT_LIGHT_BUTTON_DOWN:
+            break;
+        case EVENT_LIGHT_LONG_PRESS:
+            _beep_button(settings);
+            state->alarm_enabled = !state->alarm_enabled;
+            _running_display(event, settings, state);
+            break;
+        case EVENT_TIMEOUT:
+            movement_move_to_face(0);
+            break;
+        case EVENT_BACKGROUND_TASK:
+            if (state->alarm_enabled)
+                movement_play_alarm();
+            _schedule_alarm(settings, state);
+            break;
+        case EVENT_LOW_ENERGY_UPDATE:
+            break;
+        default:
+            return movement_default_loop_handler(event, settings);
+    }
+
+    /* Slow down frequency after first loop for snappiness */
```

Comment:
This could be increasing power consumption unnecessarily. Updating the screen when user input events arrive should make the user interface responsive and snappy without raising the tick frequency.

Reply:
I'm not sure why I set the tick frequency to 1/4 Hz on the first run of the main loop. Snappiness? I could not notice any difference in the emulator. I removed this exception and set the frequency to 1 Hz for the entire face, except for the settings loop with blinking digits. Fix is in 7e28d40.